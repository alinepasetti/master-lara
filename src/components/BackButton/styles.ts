import styled, { css } from 'styled-components';
import { BiArrowBack } from 'react-icons/bi';

export const Back = styled(BiArrowBack)`${({ theme }) => css`
  width: 40px;
  height: 40px;
  color: ${theme.colors.mediumGray};
  background-color: ${theme.colors.white};
  font-size: 0.8rem;
  border-radius: 50%;
  padding: 0.7rem;
  position: absolute;
  left: 1.6rem;
  top: 1.6rem;
`}`;
