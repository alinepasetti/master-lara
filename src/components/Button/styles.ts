import styled, { css } from 'styled-components';

export const Button = styled.a` ${({ theme }) => css`
  width: 90%;
  display: block;
  padding: 1.6rem;
  font-size: 2.4rem;
  background: ${theme.colors.primaryColor};
  color: ${theme.colors.white};
  border: none;
  margin: 0 auto 2.4rem;
  text-align: center;
  text-decoration: none;
`}`;
