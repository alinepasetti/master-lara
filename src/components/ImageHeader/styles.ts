import styled from 'styled-components';

export const Image = styled.img`
  width: 100%;
  height: 40vh;
  object-fit: cover;
  object-position: center;
`;
