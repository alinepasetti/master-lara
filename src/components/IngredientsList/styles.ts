import styled, { css } from 'styled-components';
import { AiFillCloseCircle } from 'react-icons/ai';
import { AiFillCheckCircle } from 'react-icons/ai';

type HasIngredient = {
  hasIngredient: boolean;
};

const uncheckIconUrl = (theme) =>
  `url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' fill='none' viewBox='0 0 24 24' stroke-width='1.5' stroke='%23${theme.colors.secondaryColor.substring(
    1,
  )}' class='w-6 h-6'%3E%3Cpath stroke-linecap='round' stroke-linejoin='round' d='M9.75 9.75l4.5 4.5m0-4.5l-4.5 4.5M21 12a9 9 0 11-18 0 9 9 0 0118 0z' /%3E%3C/svg%3E")`;

const checkIconUrl = (theme) =>
  `url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' fill='none' viewBox='0 0 24 24' stroke-width='1.5' stroke='%23${theme.colors.green.substring(
    1,
  )}' class='w-6 h-6'%3E%3Cpath stroke-linecap='round' stroke-linejoin='round' d='M9 12.75L11.25 15 15 9.75M21 12a9 9 0 11-18 0 9 9 0 0118 0z' /%3E%3C/svg%3E%0A")`;

export const Container = styled.ul`
  padding: 2.4rem;
  padding-top: 0;
  margin: 0;

  &> a {
    margin-top: 1.6rem;
  }
`;

export const ListItem = styled.li<HasIngredient>`${({
  theme,
  hasIngredient,
}) => css`
  list-style: none;

  &::before {
    display: inline-block;
    content: ${hasIngredient ? checkIconUrl(theme) : uncheckIconUrl(theme)};
    width: 1.8rem;
    padding-right: 1rem;
}
`}`;

export const CheckIcon = styled(AiFillCheckCircle)`${({ theme }) => css`
  color: ${theme.colors.green};
  margin-right: 2.4rem;
  height: 2.4rem;
  width: 2.4rem;
`}`;

export const UncheckIcon = styled(AiFillCloseCircle)`${({ theme }) => css`
  color: ${theme.colors.secondaryColor};
  margin-right: 2.4rem;
  height: 2.4rem;
  width: 2.4rem;
`}`;
