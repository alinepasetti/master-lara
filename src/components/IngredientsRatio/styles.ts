import styled, { css } from 'styled-components';

export const Container = styled.p`${({ theme }) => css`
  font-size: 1.6rem;
  margin: 0 0;
  color: ${theme.colors.white};
  background-color: ${theme.colors.primaryColor};
  font-size: 1.2rem;
  font-weight: 400;
  position: absolute;
  right: 1.6rem;
  top: 1.6rem;
  border-radius: 8px;
  box-shadow: 0 0 5px ${theme.colors.mediumGray};
  padding: 1rem;
`}`;
