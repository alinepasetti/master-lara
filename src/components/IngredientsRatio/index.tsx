import * as Styled from './styles';

export type IngredientsRatioProps = {
  ingredientRatio: string;
};

export const IngredientsRatio = ({
  ingredientRatio,
}: IngredientsRatioProps) => {
  return (
    <Styled.Container>
      {ingredientRatio}
      <span>ingredients</span>
    </Styled.Container>
  );
};
