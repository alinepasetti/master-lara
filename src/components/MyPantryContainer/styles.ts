import styled from 'styled-components';
import { Button as SearchButton } from '../Button/styles';

export const Container = styled.div`
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;

  ${SearchButton} {
    border-radius: 8px;
    width: 10.8rem;
  }
`;
