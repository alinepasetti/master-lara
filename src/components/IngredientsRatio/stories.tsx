import { Meta, Story } from '@storybook/react/types-6-0';
import { IngredientsRatio, IngredientsRatioProps } from '../IngredientsRatio';

export default {
  title: 'IngredientsRatio',
  component: IngredientsRatio,
  args: {
    children: 'O texto est√° escuro',
  },
  argTypes: {
    children: { type: 'string' },
  },
} as Meta;

export const Template: Story<IngredientsRatioProps> = (args) => (
  <IngredientsRatio {...args} />
);
