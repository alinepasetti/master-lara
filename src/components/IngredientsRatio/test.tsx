import { screen } from '@testing-library/react';
import { IngredientsRatio } from '../IngredientsRatio';
import { renderTheme } from '../../styles/render-theme';

describe('<IngredientsRatio />', () => {
  it('should render a heading', () => {
    renderTheme(<IngredientsRatio ingredientRatio={30} />);
    const heading = screen.getByText('30');
    expect(heading).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    renderTheme(<IngredientsRatio ingredientRatio={30} />);
    expect(screen.getByText('30')).toMatchSnapshot();
  });
});
