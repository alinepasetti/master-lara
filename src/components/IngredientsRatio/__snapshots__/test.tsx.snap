// Vitest Snapshot v1

exports[`<IngredientsRatio /> > should match snapshot 1`] = `
.c0 {
  font-size: 1.6rem;
  margin: 0 0;
  color: #FFFFFF;
  background-color: #000000;
  font-size: 1.2rem;
  font-weight: 400;
  position: absolute;
  right: 1.6rem;
  top: 1.6rem;
  border-radius: 8px;
  padding: 1rem;
}

<p
  class="c0"
>
  30
  <span>
    %
  </span>
</p>
`;
